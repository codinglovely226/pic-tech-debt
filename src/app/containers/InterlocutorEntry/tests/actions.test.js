/* eslint-disable */

import {
  addCommunication,
  replaceCommunications,
  setPastFeaturesHistory,
  setPastTreatmentHistory,
  updateCommunicationStatus,
  updateInterlocutorAddress1,
  updateInterlocutorAddress2,
  updateInterlocutorCategory,
  updateInterlocutorCity,
  updateInterlocutorContactDetail,
  updateInterlocutorCountry,
  updateInterlocutorDirection,
  updateInterlocutorGeometry,
  updateInterlocutorISOCountry,
  updateInterlocutorLocation,
  updateInterlocutorMedium,
  updateInterlocutorMediumDetail,
  updateInterlocutorName,
  updateInterlocutorOrganisation,
  updateInterlocutorPlaceId,
  updateInterlocutorPostcode,
  updateInterlocutorState,
  updateInterlocutorTelephone,
} from '../actions';

describe('App Actions', () => {
  const mockedDate = new Date(2018, 2, 4, 4, 59, 13, 805);
  const spi = {
    name: 'SPI Name',
    id: 'spi1',
    siteId: '8',
  };
  global.Date = jest.fn(() => mockedDate);
   describe('Interlocutor action creators', () => {
    it('addCommunication', () => {
      expect(addCommunication(1, spi)).toMatchSnapshot();
    });
     it('replaceCommunications', () => {
       expect(replaceCommunications(1, [{}, {}])).toMatchSnapshot();
     });
     it('setPastFeaturesHistory', () => {
       expect(setPastFeaturesHistory(1, [{}, {}])).toMatchSnapshot();
     });
     it('setPastTreatmentHistory', () => {
       expect(setPastTreatmentHistory(1, [{}, {}])).toMatchSnapshot();
     });
     it('updateCommunicationStatus', () => {
       expect(updateCommunicationStatus(1, 'status')).toMatchSnapshot();
     });
     it('updateInterlocutorAddress1', () => {
       expect(updateInterlocutorAddress1(1, 'address 1')).toMatchSnapshot();
     });
     it('updateInterlocutorAddress2', () => {
       expect(updateInterlocutorAddress2(1, 'address 2')).toMatchSnapshot();
     });
     it('updateInterlocutorCategory', () => {
       expect(updateInterlocutorCategory(1, 'category')).toMatchSnapshot();
     });
     it('updateInterlocutorCity', () => {
       expect(updateInterlocutorCity(1, 'city')).toMatchSnapshot();
     });
     it('updateInterlocutorContactDetail', () => {
       expect(updateInterlocutorContactDetail(1, 'contact detail')).toMatchSnapshot();
     });
     it('updateInterlocutorCountry', () => {
       expect(updateInterlocutorCountry(1, 'country')).toMatchSnapshot();
     });
     it('updateInterlocutorDirection', () => {
       expect(updateInterlocutorDirection(1, 'direction')).toMatchSnapshot();
     });
     it('updateInterlocutorGeometry', () => {
       expect(updateInterlocutorGeometry(1, {})).toMatchSnapshot();
     });
     it('updateInterlocutorISOCountry', () => {
       expect(updateInterlocutorISOCountry(1, 'AU')).toMatchSnapshot();
     });
     it('updateInterlocutorLocation', () => {
       expect(updateInterlocutorLocation(1, 'location')).toMatchSnapshot();
     });
     it('updateInterlocutorMedium', () => {
       expect(updateInterlocutorMedium(1, 'medium')).toMatchSnapshot();
     });
     it('updateInterlocutorMediumDetail', () => {
       expect(updateInterlocutorMediumDetail(1, 'medium detail')).toMatchSnapshot();
     });
     it('updateInterlocutorName', () => {
       expect(updateInterlocutorName(1, 'caller name')).toMatchSnapshot();
     });
     it('updateInterlocutorOrganisation', () => {
       expect(updateInterlocutorOrganisation(1, 'org')).toMatchSnapshot();
     });
     it('updateInterlocutorPlaceId', () => {
       expect(updateInterlocutorPlaceId(1, 'placeid')).toMatchSnapshot();
     });
     it('updateInterlocutorPostcode', () => {
       expect(updateInterlocutorPostcode(1, 'postcode')).toMatchSnapshot();
     });
     it('updateInterlocutorState', () => {
       expect(updateInterlocutorState(1, 'state')).toMatchSnapshot();
     });
     it('updateInterlocutorTelephone', () => {
       expect(updateInterlocutorTelephone(1, 'telephone')).toMatchSnapshot();
     });
  });
});
