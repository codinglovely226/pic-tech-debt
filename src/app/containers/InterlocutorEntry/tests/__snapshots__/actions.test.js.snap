exports[`App Actions Interlocutor action creators addCommunication 1`] = `
Object {
  "id": 1,
  "spi": Object {
    "id": "spi1",
    "name": "SPI Name",
    "siteId": "8",
  },
  "type": "pic/InterlocutorEntry/ADD_COMMUNICATION",
}
`;

exports[`App Actions Interlocutor action creators replaceCommunications 1`] = `
Object {
  "communications": Array [
    Object {},
    Object {},
  ],
  "id": 1,
  "type": "pic/InterlocutorEntry/REPLACE_COMMUNICATIONS",
}
`;

exports[`App Actions Interlocutor action creators setPastFeaturesHistory 1`] = `
Object {
  "pastFeatures": 1,
  "type": "pic/InterlocutorEntry/SET_PAST_FEATURES_HISTORY",
}
`;

exports[`App Actions Interlocutor action creators setPastTreatmentHistory 1`] = `
Object {
  "pastTreatments": 1,
  "type": "pic/InterlocutorEntry/SET_PAST_TREATMENT_HISTORY",
}
`;

exports[`App Actions Interlocutor action creators updateCommunicationStatus 1`] = `
Object {
  "id": 1,
  "immediate": false,
  "status": "status",
  "type": "UPDATE_COMMUNICATION_STATUS",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorAddress1 1`] = `
Object {
  "address1": "address 1",
  "id": 1,
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_ADDRESS1",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorAddress2 1`] = `
Object {
  "address2": "address 2",
  "id": 1,
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_ADDRESS2",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorCategory 1`] = `
Object {
  "category": "category",
  "id": 1,
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_CATEGORY",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorCity 1`] = `
Object {
  "city": "city",
  "id": 1,
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_CITY",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorContactDetail 1`] = `
Object {
  "contact": "contact detail",
  "id": 1,
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_CONTACT_DETAIL",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorCountry 1`] = `
Object {
  "country": "country",
  "id": 1,
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_COUNTRY",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorDirection 1`] = `
Object {
  "direction": "direction",
  "id": 1,
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_DIRECTION",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorGeometry 1`] = `
Object {
  "geometry": Object {},
  "id": 1,
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_GEOMETRY",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorISOCountry 1`] = `
Object {
  "id": 1,
  "isoCountry": "AU",
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_ISOCOUNTRY",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorLocation 1`] = `
Object {
  "id": 1,
  "location": undefined,
  "old": "location",
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_LOCATION",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorMedium 1`] = `
Object {
  "id": 1,
  "medium": "medium",
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_MEDIUM",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorMediumDetail 1`] = `
Object {
  "appKey": undefined,
  "detail": "medium detail",
  "id": 1,
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_MEDIUM_DETAIL",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorName 1`] = `
Object {
  "appKey": undefined,
  "id": 1,
  "name": "caller name",
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_NAME",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorOrganisation 1`] = `
Object {
  "id": 1,
  "organisation": "org",
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_ORGANISATION",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorPlaceId 1`] = `
Object {
  "id": 1,
  "placeId": "placeid",
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_PLACEID",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorPostcode 1`] = `
Object {
  "id": 1,
  "postcode": "postcode",
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_POSTCODE",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorState 1`] = `
Object {
  "id": 1,
  "state": "state",
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_STATE",
}
`;

exports[`App Actions Interlocutor action creators updateInterlocutorTelephone 1`] = `
Object {
  "id": 1,
  "telephone": "telephone",
  "type": "pic/InterlocutorEntry/UPDATE_INTERLOCUTOR_TELEPHONE",
}
`;
